pallets/nft/src/lib.rs

```diff
@@ -107,22 +163,62 @@ pub mod pallet {
 			Ok(().into())
 		}
 
-		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(2, 4))]
-		pub fn root_mint_nft(origin: OriginFor<T>, metadata: Cid, pending_nft: PendingNftOf<T>) -> DispatchResultWithPostInfo {
-			let _ = ensure_root(origin)?;
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(2, 1))]
+		pub fn cancel_nft_request(origin: OriginFor<T>, pending_nft: PendingNftOf<T>, reason: TextMessage) -> DispatchResultWithPostInfo {
+			let _account_id = ensure_signed(origin)?;
```

Comment:
Is it possible to do the same validation without assigning?

Reply:
Yeah, seems like - no warnings so far. Will remove this assignment